### PR TITLE
Add TOF limits to SCD Event Data Reduction Interface

### LINKED ISSUE #17013: SCD interface should have range of TOF to load
Range in TOF should be added to Load in SCD Interface
Default of Precount should be changed to True in hint since default is true.
### Expected behavior
Should be the same as SCD reduction scripts
event_ws = LoadEventNexus( Filename=full_name,
                           FilterByTofMin=min_tof, FilterByTofMax=max_tof )
event_ws = FilterBadPulses(InputWorkspace=event_ws, LowerCutoff = 95)
### Actual behavior
mantidplot crashes:
Running IntegrateEllipsoids
IntegrateEllipsoids-[Notice] IntegrateEllipsoids started
terminate called after throwing an instance of 'boost::exception_detail::clone_impl<boost::exception_detail::error_info_injector<boost::math::rounding_error> >'
  what():  Error in function boost::math::round<d>(d): Value -nan can not be represented in the target integer type.
Aborted (core dumped)

### Steps to reproduce the behavior
Run the SCD interface with TOPAZ_20379 and Ellipsoidal Integration option.

### PR DESCRIPTION
**Pull request by VickieLynch**
Description of work.
Added the TOF limits and FilterBadPulses used in reduction python scripts.  With DAS change, there are TOF=0 events in file that will be removed with these changes.

**To test:**

Use SCD interface in Diffraction menu and ResetDefault Settings in File menu.
Make these changes
Filename TOPAZ_20379
Number of Peaks to Find 1500
Ellipsoidal Integration with Specify Size option
Fixes #17013

_Does not need to be in the release notes._

---
#### Reviewer

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):
##### Code Review
- [ ] Is the code of an acceptable quality?
- [ ] Does the code conform to the [coding standards](http://www.mantidproject.org/Coding_Standards)? Is it well structured with small focussed classes/methods/functions?
- [ ] Are there unit/system tests in place? Are the unit tests small and test the a class in isolation?
- [ ] If there are changes in the release notes then do they describe the changes appropriately?
##### Functional Tests
- [ ] Do changes function as described? Add comments below that describe the tests performed?
- [ ] How do the changes handle unexpected situations, e.g. bad input?
- [ ] Has the relevant documentation been added/updated?
- [ ] Is user-facing documentation written in a user-friendly manner?
- [ ] Has developer documentation been updated if required?
- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: MantidQt/CustomInterfaces/src/MantidEVWorker.cpp [hunk 1]
```diff
@@ -156,12 +156,21 @@ bool MantidEVWorker::loadAndConvertToMD(
...
+      alg->setProperty("FilterByTofMax", 16666.0);
       alg->setProperty("LoadMonitors", true);
 
       if (!alg->execute())
         return false;
 
+      alg = AlgorithmManager::Instance().create("FilterBadPulses");
+      alg->setProperty("InputWorkspace", ev_ws_name);
+      alg->setProperty("OutputWorkspace", ev_ws_name);
+      alg->setProperty("LowerCutoff", 95.0);
```
Comment (peterfpeterson) [inline MantidQt/CustomInterfaces/src/MantidEVWorker.cpp]: New wisdom on this is to set the value to something significantly lower so you measure during beam coming online. The value POWGEN uses is 25.

File: MantidQt/CustomInterfaces/src/MantidEVWorker.cpp [hunk 2]
```diff
@@ -156,12 +156,21 @@ bool MantidEVWorker::loadAndConvertToMD(
       IAlgorithm_sptr alg = AlgorithmManager::Instance().create("Load");
       alg->setProperty("Filename", file_name);
       alg->setProperty("OutputWorkspace", ev_ws_name);
-      alg->setProperty("Precount", true);
+      alg->setProperty("FilterByTofMin", 1000.0);
```
Comment (peterfpeterson) [inline MantidQt/CustomInterfaces/src/MantidEVWorker.cpp]: Is it save to hard code the tof range?

Comment (VickieLynch) [inline MantidQt/CustomInterfaces/src/MantidEVWorker.cpp]: I could add limits to interface, but this is something they have never wanted to change until these TOF=0's appeared. I will use your cutoff suggestion.

**Comment by peterfpeterson** [comment]
LGTM :shipit: once the builds are passing.

**Comment by AndreiSavici** [comment]
1000 microseconds corresponds to something like 23 inverse angstoms momentum (k)

**Pull request merged**